### PR TITLE
6852 unfinalized steps styling in loan calendar

### PR DESCRIPTION
**Pull request by emitche**
- Style draft steps inside the loan calendar
- Ensure colors chosen are color contrast accessible
- Adjust the calendar legend

### CODE REVIEW

File: app/assets/stylesheets/admin/_colors.scss [hunk 1]
```diff
@@ -26,17 +26,19 @@ $background-gray: rgb(243, 243, 243);
...
+$gray8: #545454;  // Meets accessibility standard on $table-gray
+$gray9: #595959; // Meets accessibility standards on white background
 $body: #333;
 $black: black;
 
 // Color values from project step model
-$onTime: hsl(120, 73%, 57%);
-$superEarly: hsl(120, 41%, 47%);
-$barelyLate: hsl(56, 100%, 66%);
-$superLate: hsl(0, 74%, 54%);
```
Comment (emitche) [inline app/assets/stylesheets/admin/_colors.scss]: hsl is a bit non-standard in terms of using web development tools. Having the rgb values available makes things easier for future styling.

File: app/assets/stylesheets/admin/_variables.scss [hunk 2]
```diff
@@ -3,3 +3,9 @@
 $content-wrapper-padding: 20px;
 $sidebar-width: 190px;
 $app-fonts: 'Open Sans', sans-serif;
+
+// Common styles
```
Comment (emitche) [inline app/assets/stylesheets/admin/_variables.scss]: Is there a better place to put things if using `@extend`?

File: app/assets/stylesheets/admin/_variables.scss [hunk 3]
```diff
@@ -3,3 +3,9 @@
...
+.draft {
```
Comment (emitche) [inline app/assets/stylesheets/admin/_variables.scss]: Intended to dry up step draft styles. We use something similar in both the timeline in the calendar. If we change either in the future, the change can touch both places at once.

File: app/assets/stylesheets/admin/loans/timeline/_table.scss [hunk 4]
```diff
@@ -1,4 +1,5 @@
 @import "../../colors";
+@import "../../variables";
```
Comment (emitche) [inline app/assets/stylesheets/admin/loans/timeline/_table.scss]: I'd ideally like to get away from importing variables and colors. Fixing the underlying bug that requires this as a fix is part of our backlog.

File: app/views/admin/calendar/_legend.html.slim [hunk 5]
```diff
@@ -4,20 +4,23 @@ div#legend-content
       div.column.col-lg-6
         table.legend-block
           tr
+            td
```
Comment (emitche) [inline app/views/admin/calendar/_legend.html.slim]: Draft steps do not exist in the main calendar. Do we need to have a revised list of legend items for the general calendar or is that not needed?

File: app/assets/javascripts/backbone/views/calendar_view.coffee [hunk 6]
```diff
@@ -29,6 +29,12 @@ class MS.Views.CalendarView extends Backbone.View
   # Load custom content inside the fullCalendar rendered event rather than title only
   eventAfterRender: (calEvent, initialElement) ->
     $(initialElement).find('.fc-content').append(calEvent.html)
+
+    if calEvent.is_finalized
+      $(initialElement).addClass('finalized')
+    else
+      $(initialElement).addClass('draft')
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/calendar_view.coffee]: Given that CalendarEventSerializer already has the event_classes method, perhaps better to put this logic in there?

File: app/assets/stylesheets/admin/_colors.scss (hunk 1 above)
Comment (hooverlunch) [reply app/assets/stylesheets/admin/_colors.scss]: Is there any reason not to just convert to RGB? I would be fine with that.

File: app/assets/stylesheets/admin/_variables.scss (hunk 3 above)
Comment (hooverlunch) [reply app/assets/stylesheets/admin/_variables.scss]: Great idea! If you intend for this style to just be applicable to project steps, It should maybe go in the project steps folder? Or, if it's going to be an app wide style for draft-type things, Then maybe it should go in _general.scss?

File: app/assets/stylesheets/admin/_variables.scss [hunk 7]
```diff
@@ -1,5 +1,13 @@
+@import "colors";
+
 // Common variables
 
 $content-wrapper-padding: 20px;
 $sidebar-width: 190px;
 $app-fonts: 'Open Sans', sans-serif;
+
```
Comment (hooverlunch) [inline app/assets/stylesheets/admin/_variables.scss]: Whoops, I accidentally deleted your comment here. See below.

File: app/assets/stylesheets/admin/calendar/_calendar.scss [hunk 8]
```diff
@@ -79,6 +79,12 @@
     }
   }
 
+  .cal-step {
+    &.draft {
+      @extend .draft;
+    }
```
Comment (hooverlunch) [inline app/assets/stylesheets/admin/calendar/_calendar.scss]: I'm not sure that you need to use @extend here or below. If you define a global class, it should Already apply to any Element with that class, right? Try it without?

File: app/assets/stylesheets/admin/loans/timeline/_table.scss (hunk 4 above)
Comment (hooverlunch) [reply app/assets/stylesheets/admin/loans/timeline/_table.scss]: Amen!

File: app/views/admin/calendar/_legend.html.slim (hunk 5 above)
Comment (hooverlunch) [reply app/views/admin/calendar/_legend.html.slim]: I don't think it's a big deal.

File: app/assets/javascripts/backbone/views/calendar_view.coffee (hunk 6 above)
Comment (emitche) [reply app/assets/javascripts/backbone/views/calendar_view.coffee]: I tried, but was having trouble with it working, so this was the best solution I could come up with. I'm not sure why it was not working.

File: app/assets/stylesheets/admin/calendar/_calendar.scss (hunk 8 above)
Comment (emitche) [reply app/assets/stylesheets/admin/calendar/_calendar.scss]: That makes sense. I overcomplicated things I think.

File: app/views/admin/calendar/_legend.html.slim (hunk 5 above)
Comment (emitche) [reply app/views/admin/calendar/_legend.html.slim]: Ok.

File: app/assets/javascripts/backbone/views/calendar_view.coffee (hunk 6 above)
Comment (emitche) [reply app/assets/javascripts/backbone/views/calendar_view.coffee]: I figured out what I needed to change. Fixed. Thanks!

Comment (emitche) [reply app/assets/javascripts/backbone/views/calendar_view.coffee]: Update: The problem is that the event serializer does not successfully distinguish between finalized and draft steps. I'm not sure why that happens. But finalized steps get the `.draft` class though they are not drafts.

**Comment by emitche** [comment]
This has received code review and just needs a QA review. Thank you, @linesarefuzzy!

File: app/assets/stylesheets/admin/_colors.scss (hunk 1 above)
Comment (linesarefuzzy) [reply app/assets/stylesheets/admin/_colors.scss]: The reason I used HSL here is that the project step model does calculations that require colors in HSL format. The colors are defined in that model as well as here. I think it's useful to have them defined in the same way in both places.

**Comment by emitche** [comment]
I'll take a look. Thanks!

**Comment by emitche** [comment]
@linesarefuzzy: The issue you brought up during QA has been addressed. Is anything else needed?

**Comment by linesarefuzzy** [comment]
Yes, thank you! However, I just did another QA review and noticed that incomplete steps are not being bolded, yet the legend says they are. It's that way in develop too. I'll make an issue.

**Comment by emitche** [comment]
Yes, I'll make a ticket for that. Thanks for letting me know.

**Comment by emitche** [comment]
I'll merge this into develop.

File: app/assets/stylesheets/admin/_colors.scss (hunk 1 above)
Comment (emitche) [reply app/assets/stylesheets/admin/_colors.scss]: I'll keep the values in hsl but keep rgb in the comments.

**Pull request merged**